app/views/finders/_option_select_facet.html.erb

```diff
@@ -1,12 +1,14 @@
 <% cache_if option_select_facet.cacheable?, option_select_facet.cache_key do %>
-  <% close_facet = option_select_facet.unselected? && ((option_select_facet_counter > 0) || option_select_facet.close_facet?) %>
   <%= render partial: 'components/option-select', locals: {
     :key => option_select_facet.key,
     :title => option_select_facet.name,
     :aria_controls_id => "js-search-results-info",
     :options_container_id => option_select_facet.key,
     :options => option_select_facet.options("js-search-results-info", option_select_facet.key),
-    :closed_on_load => close_facet,
+    # on desktop, if it's unselected and it's not the first facet, close it.
+    :closed_on_desktop => option_select_facet.unselected? && option_select_facet_counter > 0,
+    # on mobile, if it's unselected and it's not the first facet, or if it IS the first (and unselected) facet but contains more than 10 options, close it.
+    :closed_on_mobile => option_select_facet.unselected? && (option_select_facet_counter > 0 || (option_select_facet_counter == 0 && option_select_facet.allowed_values.count > 10)),
```

Comment:
As with the previous comment, rather than having a lot of logic in the view, which is much harder to test, I would expand `closed_on_load` to handle the different conditions for desktop and mobile. That way, the view would remain virtually untouched.